lib/Controller/ScrobblerController.php

```diff
@@ -0,0 +1,126 @@
+<?php declare(strict_types=1);
+
+/**
+ * ownCloud - Music app
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Matthew Wells
+ * @copyright Matthew Wells 2025
+ */
+
+namespace OCA\Music\Controller;
+
+use OCA\Music\Service\ScrobbleServiceException;
+use OCA\Music\Service\ExternalScrobbler;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\AppFramework\Http\StandaloneTemplateResponse;
+use OCP\IL10N;
+use OCP\IRequest;
+
+class ScrobblerController extends Controller {
+	private IL10N $l10n;
+
+	private ?string $userId;
+
+	/** @var ExternalScrobbler[] $externalScrobblers */
+	private array $externalScrobblers;
+
+	public function __construct(string $appName,
+								IRequest $request,
+								IL10N $l10n,
+								?string $userId,
+								array $externalScrobblers) {
+		parent::__construct($appName, $request);
+		$this->l10n = $l10n;
+		$this->userId = $userId;
+		$this->appName = $appName;
+		$this->externalScrobblers = $externalScrobblers;
+	}
+
+	/**
+	 * @PublicPage
```

Comment:
This and `clearSession` shouldn't have the `@PublicPage` annotation because that indicates an endpoint which can be called without a logged-in user (like on a public share page). Instead, there should be `@NoAdminRequired` to allow calling for any logged-in user (the default is to allow only admin users). And in case you wander, the entry points of `AmpacheController` and `SubsonicController` are marked as "public" because both of these have their own user credential handling, independent from the Nextcloud user sessions.

The `@NoSameSiteCookieRequired` is probably necessary here as written, since this endpoint is called by the Last.fm site. But in the function `clearSession`, it shouldn't be needed.